### PR TITLE
add InternalError to values returned by SubmissionResult

### PR DESCRIPTION
**Pull request by mziolekda**
### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/unreleased.rst), if appropriate
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Comment by dasormeter** [comment]
@mziolekda @dajmaki actually was just looking at related matters regarding loss of fidelity of response we will lose at submission time by having write path submission only, and feedback coming from read path:

A few specific things I wanted to get your thoughts on:
1) Where we check for empty archive and reject for package submission would it similarly make sense for this to get an error SubmissionResult saying "archive empty or invalid" or do you think we should leave all validation to the processing of the submission and only return the error and reason for error in the result read path? 

2) Presuming for invalid name submission the command would return as Acknowledged but it would only be in reading reject message from read path you would see the error detail (after async refactors)

3) How granular do we want to go with these internal errors?  @mziolekda  what kind of internal errors did you have in mind that prompted this PR?

**Comment by dasormeter** [comment]
> @mziolekda @dajmaki actually was just looking at related matters regarding loss of fidelity of response we will lose at submission time by having write path submission only, and feedback coming from read path:
> 
> A few specific things I wanted to get your thoughts on:
> 
> 1. Where we check for empty archive and reject for package submission would it similarly make sense for this to get an error SubmissionResult saying "archive empty or invalid" or do you think we should leave all validation separate to the command submission and only return the error and reason for error in the reject event in the read path?
> 2. Presuming for invalid name submission the command would return as Acknowledged but it would only be in reading reject message from read path you would see the error detail (after async refactors)
> 3. How granular do we want to go with these internal errors?  @mziolekda  what kind of internal errors did you have in mind that prompted this PR?

actually looks like the string validation is a daml_lf utility function hint.map(p => Party.fromString(p)) match {

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/services/ApiSubmissionService.scala [hunk 1]
```diff
@@ -118,6 +119,11 @@ class ApiSubmissionService private (
             logger.debug(s"Submission of command {} was not supported", commands.commandId.unwrap)
             Failure(Status.INVALID_ARGUMENT.asRuntimeException)
 
+          case Success(InternalError(reason)) =>
+            logger.debug(
+              s"Submission of command ${commands.commandId.unwrap} failed, reason=$reason ")
+            Failure(Status.INTERNAL.asRuntimeException)
```
Comment (stefanobaghino-da) [inline ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/services/ApiSubmissionService.scala]:
```suggestion
            Failure(Status.INTERNAL.augmentDescription(reason).asRuntimeException)
```

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/services/ApiSubmissionService.scala [hunk 2]
```diff
@@ -118,6 +119,11 @@ class ApiSubmissionService private (
             logger.debug(s"Submission of command {} was not supported", commands.commandId.unwrap)
             Failure(Status.INVALID_ARGUMENT.asRuntimeException)
 
+          case Success(InternalError(reason)) =>
+            logger.debug(
+              s"Submission of command ${commands.commandId.unwrap} failed, reason=$reason ")
```
Comment (stefanobaghino-da) [inline ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/services/ApiSubmissionService.scala]:
```suggestion
              s"Submission of command ${commands.commandId.unwrap} failed due to an internal error, reason=$reason ")
```

**Comment by mziolekda** [comment]
> @mziolekda @dajmaki actually was just looking at related matters regarding loss of fidelity of response we will lose at submission time by having write path submission only, and feedback coming from read path:
> 
> A few specific things I wanted to get your thoughts on:
> 
> 1. Where we check for empty archive and reject for package submission would it similarly make sense for this to get an error SubmissionResult saying "archive empty or invalid" or do you think we should leave all validation separate to the command submission and only return the error and reason for error in the reject event in the read path?
Write path of the participant state should be a shallow adaptation layer that doesn't perform any validations. Validations belong on 2 levels: a) in ledger api services where the client's gRPC request is first scrutinized (and extending into DAMLe interpretation to weed out nonsensical requests) b) in the validations performed as part of the consensus layer, result of those is communicated through the read path.

> 2. Presuming for invalid name submission the command would return as Acknowledged but it would only be in reading reject message from read path you would see the error detail (after async refactors)
Correct, consensus layer validations are reported through read path.

> 3. How granular do we want to go with these internal errors?  @mziolekda  what kind of internal errors did you have in mind that prompted this PR?
We need internal errors to be able to report back to client all kinds of critical errors such as lost connection, a crash, lack of memory etc.

**Pull request merged**